### PR TITLE
Fact 1584 - Send Letter Service - Upgrade to Spring Boot 3 

### PR DESCRIPTION
**Pull request by RuthKirby**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FACT-1584

### Change description ###
Implementation decisions

- Tracing Configuration's SensitiveHeadersRequestTraceFilter has been removed as it comes from a project incompatible with Spring Boot 3 (SB3). Tracing has [completely changed](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#httptrace-endpoint-renamed-to-httpexchanges) in this version of SB so will need to be rewritten (maybe in this PR?)
- The MockFilterConfigs from an incompatible version of Microsoft Application Insights has been removed as they are probably not needed
- JsonType Hibernate Type for Postgres SQL [can handle JsonBinaryType as well](https://vladmihalcea.com/how-to-map-json-objects-using-generic-hibernate-types/) so I have used JsonType when updating entities for Hibernate 6
- This PR uses a snapshot of Apache Mail as Apache Mail has [not yet released](https://lists.apache.org/thread/prgf2m77oqwvbhgtzgry6h9x7pw5oz1d) a version compatible with SB3. Should release "soon"

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
Potentially
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sendletter/tasks/reports/EmailSenderTest.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package uk.gov.hmcts.reform.sendletter.tasks.reports;
 
 import com.icegreen.greenmail.util.ServerSetupTest;
-import org.apache.commons.mail.util.MimeMessageParser;
+import jakarta.activation.DataSource;
+import jakarta.mail.Address;
+import jakarta.mail.internet.MimeMessage;
+//import org.apache.commons.mail.util.MimeMessageParser;
```
Comment (joshblackmoor) [inline src/test/java/uk/gov/hmcts/reform/sendletter/tasks/reports/EmailSenderTest.java:7]: Remove unneeded :)

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -21,7 +21,7 @@ secrets = [
     secret('ftp-public-key', 'TEST_FTP_PUBLIC_KEY'),
     secret('xerox-ftp-hostname', 'TEST_FTP_HOSTNAME'),
     secret('launch-darkly-sdk-key', 'LAUNCH_DARKLY_SDK_KEY'),
-    secret('launch-darkly-offline-mode', 'LAUNCH_DARKLY_OFFLINE_MODE')
+    secret('launch-darkly-offline-mode', 'LAUNCH_DARKLY_OFFLINE_MODE'),
```
Comment (joshblackmoor) [inline Jenkinsfile_CNP:24]: don't need

**Review by joshblackmoor — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sendletter/tasks/reports/EmailSenderTest.java (hunk 1 above)
Comment (RuthKirby) [reply src/test/java/uk/gov/hmcts/reform/sendletter/tasks/reports/EmailSenderTest.java]: Import has been [removed](https://github.com/hmcts/send-letter-service/pull/2486/commits/8e1f282389dc342d153729b1cd367ee8443fb818).

File: Jenkinsfile_CNP (hunk 2 above)
Comment (RuthKirby) [reply Jenkinsfile_CNP]: Comma has been [removed](https://github.com/hmcts/send-letter-service/pull/2486/commits/b7f94dfb4584c312f5535efe8c22604eef69215f)

**Review by joshblackmoor — APPROVED** [review]

**Pull request merged**